dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,46 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+dashboard.header.getSummarizedProgressForAnonymousUser = function() {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';')
+
+  for(var i = 0; i < cookie.length; i++) {
+    var key = cookie[i].split('=')[0].trim().toLowerCase();
+
+    switch(key) {
+      case 'progress':
+        var allLevelsProgress = JSON.parse(decodeURIComponent(pair[1]));
+        for(var level in allLevelsProgress) {
+          if (allLevelsProgress.hasOwnProperty(level)) {
+            var individualLevelProgressDescription = '';
+            var individualLevelProgress = allLevelsProgress[level];
+
+            //TODO: Find a way to use the constants defined somewhere
+            if (individualLevelProgress >= 1000) {
+              individualLevelProgressDescription = 'submitted';
+            } else if (individualLevelProgress >= 30) {
+              individualLevelProgressDescription = 'perfect';
+            } else if (individualLevelProgress >= 20) {
+              individualLevelProgressDescription = 'passed';
+            } else {
+              individualLevelProgressDescription = 'attempted';
+            }
+
+            levelProgress[level] = {};
+            levelProgress[level]['status'] = individualLevelProgressDescription;
```

Comment:
We prefer the form levelProgress[level].status. In apps and shared we lint for this. Unfortunately, we don't (yet) have linting for dashboard JS.